src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  static removeDeadAnimal() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    super.hidden = hidden;
```

Comment:
[CODE QUALITY] The `super` keyword is used to call functions on an object's parent. However, here it is incorrectly used to set a property. Instead, you should assign the `hidden` property directly to `this` as `this.hidden = hidden;`.